Unit testing for Project Team Members router

jng34 opened the pull request:
Fixes #1892 

### What changes did you make and why did you make them ?
  - wrote unit tests for `projectTeamMembers.router`
  - edited the logic in the GET API route `/project/:id/:userId` in `projectTeamMembers.router.js` that incorrectly threw the HTTP code `400` when it should have been `200`, which affected the unit test for that route

### Screenshots of Passed Tests
<img width="500" alt="image" src="https://github.com/user-attachments/assets/73ca8960-fd30-46de-a157-6e8d3ae8b6b9" />

jng34 committed:
wrote unit tests for project team members router, fixed bug in GET /project/:id/:userId route

JackHaeg commented:
@jng34 It appears there are some conflicts on this PR that need to be resolved. Can you please take a look at these when you have a moment?

trillium committed:
fix: correct empty array check in projectTeamMembers router

- .find() returns an array, not an object. Changed check from
  !teamMember.projectId to proper empty array check
- Remove unused err parameter in catch block

trillium reviewed (approved):
## Review: Approve

**Summary:** Adds unit tests for Project Team Members router with good coverage of CRUD operations.

### Changes Reviewed
- ✅ `projectTeamMembers.router.test.js`: 7 tests covering GET, POST, PATCH
- ✅ `projectTeamMembers.router.js`: Bug fix + formatting

### Bug Fixed
Fixed incorrect empty array check:
```js
// Before (broken - arrays don't have .projectId)
if (\!teamMember.projectId)

// After (correct)
if (\!teamMember || teamMember.length === 0)
```

### Tests
All 7 tests pass ✅

Rebased onto upstream/development and fixed bug.

trillium committed:
Merge branch 'development' into unitTestingForProjectTeamMembers
